aten/src/ATen/native/RNN.cpp

```diff
@@ -420,7 +480,95 @@ struct LSTMCell : Cell<std::tuple<Tensor, Tensor>, cell_params> {
     auto hy = outgate * cy.tanh();
     return std::make_tuple(std::move(hy), std::move(cy));
   }
+};
+
+struct qLSTMCell : Cell<std::tuple<Tensor, Tensor>, QuantizedCellParamsStatic> {
+  using hidden_type = std::tuple<Tensor, Tensor>;
 
+  // constants for different ranges.
+  static constexpr float kQ8BitScaleWithRange2 = 2.0f / 256.0f;
+  static constexpr float kQ8BitScaleWithRange16 = 16.0f / 256.0f;
+  static constexpr float kQ32BitScaleWithRange256 = 256.0f / uint32_t(-1);
+  static constexpr int32_t kQZeroPoint = 0;
+
+  hidden_type operator()(
+      const Tensor& input,
+      const hidden_type& hidden,
+      const QuantizedCellParamsStatic& params,
+      bool pre_compute_input = false) const override {
+    const auto& hx = std::get<0>(hidden);
+    const auto& cx = std::get<1>(hidden);
```

Comment:
We're pretty deep in type indirection here (auto -> tuple contained type -> using declaration`. Can we just say `const Tensor&` here?